system/Helpers/url_helper.php

```diff
@@ -26,17 +26,24 @@
      *
      * @internal Outside the framework this should not be used directly.
      *
-     * @param string $relativePath May include queries or fragments
+     * @param array|string $relativePath URI string or array of URI segments.
+     *                                   May include queries or fragments.
+     * @param App|null     $config       Alternative Config to use
      *
      * @throws HTTPException            For invalid paths.
      * @throws InvalidArgumentException For invalid config.
      */
-    function _get_uri(string $relativePath = '', ?App $config = null): URI
+    function _get_uri($relativePath = '', ?App $config = null): URI
     {
-        $config ??= config('App');
+        $appConfig = null;
+        if ($config === null) {
+            /** @var App $appConfig */
+            $appConfig = config('App');
+        }
 
-        if ($config->baseURL === '') {
-            throw new InvalidArgumentException('_get_uri() requires a valid baseURL.');
```

Comment:
Is there a reason why we're removing this check?

Reply:
I missed it. I will check.

Reply:
Added above.

Reply:
Thanks!